### PR TITLE
Filter create command flags by driver

### PR DESCRIPTION
**Pull request by djfarrelly**
This implements the idea proposed in #776. When specifying a driver w/ the `-d` flag but not specifying a name, the command help only displays relevant flags to use with that specified driver. Example:

```
$ docker-machine create -d virtualbox
NAME:
   create - Create a machine

USAGE:
   command create [command options] [arguments...]

OPTIONS:
   --virtualbox-boot2docker-url 	The URL of the boot2docker image. Defaults to the latest available version [$VIRTUALBOX_BOOT2DOCKER_URL]
   --virtualbox-cpu-count "-1"		number of CPUs for the machine (-1 to use the number of CPUs available) [$VIRTUALBOX_CPU_COUNT]
   --virtualbox-disk-size "20000"	Size of disk for host in MB
   --virtualbox-memory "1024"		Size of memory for host in MB

Run 'docker-machine create --help' for all options.

FATA[0000] You must specify a machine name
```

**Note:** I'm very new to writing Go and the Docker machine project so please do let me know if I can make any adjustments to improve my change. Thanks!

### CODE REVIEW

**Comment by ehazlett** [comment]
Thanks for the contribution!  I think I would like to look into the codegangsta subcommands (actually sub-sub commands) instead of the regex.  Would you want to look into that?  I think it would be cleaner and perhaps less chance of breakage.  Thanks!

**Comment by djfarrelly** [comment]
Hey @ehazlett, I appreciate the review! I do agree the regex is not ideal. That's a great idea, I think it might take moving the driver flag and driver-specific flags to a sub-command rather than append all of the driver flag and driver-specific flags to the "create" command. I'd be happy to look further into this! Will report back when I have a new approach.

**Comment by ehazlett** [comment]
Thanks!

**Comment by hairyhenderson** [comment]
Nice, @djfarrelly! Is it possible to also display "global" flags? For example, stuff like `--swarm-discovery` would be useful to see all the time, regardless of which driver's being used.

**Comment by ehazlett** [comment]
@hairyhenderson i think this works with codegangsta

**Comment by djfarrelly** [comment]
**tl;dr** - Using [subcommands](https://github.com/codegangsta/cli#subcommands) turned out to be a log bigger change than I originally thought. Instead I've created a non-regex way to filter the commands and updated this PR.

The new changes will also allow the Swarm flags to remain on the filtered list as proposed by @hairyhenderson :)

-

I spent a bit of time researching & working on Subcommands and ran into an issue:

In order to nest flags with [codegangsta's cli](https://github.com/codegangsta/cli#subcommands), I believe we'd need to make each driver it's own command to attach each driver's respective flags to. The main usage would change to `docker-machine create virtualbox myname`. This feels a lot different as *virtualbox* would be the command. As a side effect each driver could potentially have more simplified flags like `--cpu-count` instead of `--vmwarevcloudair-cpu-count`. Example output:

```
$ docker-machine create virtualbox --help
NAME:
   virtualbox - Create a machine using the virtualbox driver

USAGE:
   command virtualbox [command options] [arguments...]

OPTIONS:
   --virtualbox-memory "1024"		Size of memory for host in MB
   --virtualbox-cpu-count "-1"		number of CPUs for the machine (-1 to use the number of CPUs available) [$VIRTUALBOX_CPU_COUNT]
   --virtualbox-disk-size "20000"	Size of disk for host in MB
   --virtualbox-boot2docker-url 	The URL of the boot2docker image. Defaults to the latest available version [$VIRTUALBOX_BOOT2DOCKER_URL]
```
**Bottom line:** This feels like a much bigger change than a basic filter on flags. I'm not sure it's worth that fundamental change to cleanup the flag list, but I don't have the experience contributing to this and other Docker projects to make that call!

Would love any feedback or thoughts you had @ehazlett. Thanks so much for your time!

**Comment by hairyhenderson** [comment]
@djfarrelly - The current commit looks good to me.

I'm on the fence about going to subcommands - you're right that it's a big change. A potential command line could look like:

```
$ docker-machine create --swarm-master --swarm-discovery token://abcd1234 \
  virtualbox --memory 2048 --cpu-count 4 swarm-master
```

On the one hand, it's nice to have the non-driver-specific `create` options isolated from the driver-specific options. And it involves less typing if the flags can drop the driver name.

On the other hand, it's a big change ;)

**Comment by ehazlett** [comment]
@djfarrelly thanks for the write up and great feedback!  i agree that would be a much larger change and affect the UX quite a bit.  This might not be bad when we move to a plugin model where the subcommand would actually be desired.  For now, I will do some testing around this and filtering sounds good to me given we have 100s of options in total :)

/cc @nathanleclaire thoughts?

**Comment by djfarrelly** [comment]
Awesome thoughts and feedback @hairyhenderson and @ehazlett! Agree it could be pretty cool w/ a plugin model :)

**Comment by nathanleclaire** [comment]
I really like this.  There are a few nitpicky things that I want to change before merging and it might just be easier for me to make a PR on your repo (@djfarrelly), so stay tuned.

**Comment by djfarrelly** [comment]
Sounds great @nathanleclaire, thanks for taking a look at this! I'm new to Go and contributing to Docker projects, looking forward to see what changes you're planning to make :)

**Comment by djfarrelly** [comment]
Are there any rebasing or tweaks I should make for this to be ready?

**Comment by ehazlett** [comment]
@nathanleclaire ping

**Comment by nathanleclaire** [comment]
LGTM

**Comment by nathanleclaire** [comment]
Let's please merge https://github.com/djfarrelly/machine/pull/3 before merging this PR though @djfarrelly thanks!

**Comment by djfarrelly** [comment]
:+1: https://github.com/djfarrelly/machine/pull/3 is merged

**Comment by ehazlett** [comment]
LGTM

**Comment by djfarrelly** [comment]
Thanks guys! Glad I could contribute a bit :smile:

**Comment by ehazlett** [comment]
@djfarrelly you are welcome :)  thanks for contributing!
